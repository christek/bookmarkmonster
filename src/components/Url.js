// @flow

import React from 'react';

import type { Text } from '../types/todos';

export type Props = {
  text: Text
};

const Todo = ({ text }: Props) => (
  <li>
    {text}
  </li>
);

export default Todo;
