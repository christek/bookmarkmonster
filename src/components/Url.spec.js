// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Url from './Url';

const setup = (setupProps = {}) => {
  const defaultProps = {
    text: 'Test todo',
    onClick: jest.fn()
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <Url
      text={props.text}
      onClick={props.onClick}
    />
  );

  return {
    props,
    wrapper
  };
};

describe('Url', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
  
});
