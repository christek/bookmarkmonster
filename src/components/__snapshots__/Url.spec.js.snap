// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Url calls onClick when clicked 1`] = `ShallowWrapper {}`;

exports[`Url renders without crashing 1`] = `ShallowWrapper {}`;
